.github/workflows/comment-pr.yml

```
@@ -0,0 +1,56 @@
+name: Comment on the pull request
+
+on:
+  workflow_run:
+    workflows: ["free-programming-books-lint"]
+    types:
+      - completed
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  upload:
```

```suggestion
  upload:
    permissions:
      pull-requests: write
```

This job would need to explicitly declare write access for `pull-request` here, I think?
Either that, or in the repo settings someone would have to enable write access by default for all workflows, but that's probably not a good idea for a public repo.